### PR TITLE
Java 11 (patterns with m)

### PR DESCRIPTION
**Pull request by anuragagarwal561994**
Covers the following patterns:

- marker
- master-worker-pattern
- mediator
- memento
- model-view-controller
- model-view-presenter
- module
- monad
- monostate
- multiton
- mute-idiom
- mutex

### CODE REVIEW

File: marker/src/main/java/App.java [hunk 1]
```diff
@@ -46,22 +43,10 @@
...
-      guard.enter();
-    } else {
-      logger.info("You have no permission to enter, please leave this area");
-    }
-
-    if (thief instanceof Permission) {
-      thief.steal();
-    } else {
-      thief.doNothing();
-    }
```
Comment (iluwatar) [inline marker/src/main/java/App.java:64]: The point of the design pattern is to demonstrate marker interfaces, so I think you cannot delete the `instanceof` checks here

Comment (anuragagarwal561994) [reply marker/src/main/java/App.java:64]: Pushed the changes

File: marker/src/main/java/App.java [hunk 2]
```diff
@@ -46,19 +46,20 @@
...
     if (guard instanceof Permission) {
       guard.enter();
     } else {
       logger.info("You have no permission to enter, please leave this area");
     }
 
+    //noinspection ConstantConditions
     if (thief instanceof Permission) {
-      thief.steal();
+      thief.doNothing();
```
Comment (iluwatar) [inline marker/src/main/java/App.java:62]: With this change the if and else branches do the same thing. I would leave it to `thief.steal()` as it was.

**Comment by iluwatar** [comment]
@anuragagarwal561994 The pull request has remained inactive and is about to be closed. Please comment if you're still working on it.

**Comment by anuragagarwal561994** [comment]
I will resolve it tonight

File: marker/src/main/java/App.java (hunk 2 above)
Comment (anuragagarwal561994) [reply marker/src/main/java/App.java]: Pushed the changes

**Comment by anuragagarwal561994** [comment]
@iluwatar all the issues have been resolved, please do have a look at it

File: memento/src/main/java/com/iluwatar/memento/Star.java [hunk 3]
```diff
@@ -71,7 +71,7 @@ public void timePasses() {
 
   StarMemento getMemento() {
 
-    StarMementoInternal state = new StarMementoInternal();
+    var state = new StarMementoInternal();
```
Comment (iluwatar) [inline memento/src/main/java/com/iluwatar/memento/Star.java:74]: For Memento pattern we need to mirror the changes in the `README.md` explanation

**Comment by anuragagarwal561994** [comment]
Done with the changes

**Comment by iluwatar** [comment]
@anuragagarwal561994 Many thanks for this big contribution! 👍🏻

**Comment by iluwatar** [comment]
#984 and #987

**Pull request merged**